src/frontend/src/pages/TeamsPage/TeamMembersView.tsx

```diff
@@ -0,0 +1,123 @@
+import PageBlock from '../../layouts/PageBlock';
+import { Autocomplete, Button, Grid, IconButton, TextField, useTheme } from '@mui/material';
+import { fullNamePipe } from '../../utils/pipes';
+import { Team } from 'shared';
+import { Edit } from '@mui/icons-material';
+import { useState } from 'react';
+import { useAuth } from '../../hooks/auth.hooks';
+import { useAllUsers } from '../../hooks/users.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { User } from 'shared';
+import { useSetTeamMembers } from '../../hooks/teams.hooks';
+
+interface TeamMembersViewProps {
+  team: Team;
+}
+
+const TeamMembersView: React.FC<TeamMembersViewProps> = ({ team }) => {
+  const auth = useAuth();
+  const theme = useTheme();
+  const [isEditingMembers, setIsEditingMembers] = useState(false);
+  const [members, setMembers] = useState(team.members);
+
+  const { isLoading, isError, error, data } = useAllUsers();
+  const {
+    isLoading: memberIsLoading,
+    isError: memberIsError,
+    error: memberError,
+    mutateAsync
+  } = useSetTeamMembers(team.teamId);
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (memberIsError) return <ErrorPage message={memberError?.message} />;
+  if (isLoading || memberIsLoading || !data) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    await mutateAsync(members.map((member) => member.userId));
+    setIsEditingMembers(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+  const userToAutocompleteOption = (user: User): { label: string; id: number } => {
+    return { label: `${fullNamePipe(user)}`, id: user.userId };
+  };
+
+  if (!isEditingMembers)
+    return (
+      <PageBlock
+        title={'People'}
+        headerRight={<IconButton disabled={hasPerms()} onClick={() => setIsEditingMembers(true)} children={<Edit />} />}
+      >
+        <Grid container spacing={1}>
+          <Grid item xs={12} md={12}>
```

Comment:
if xs={12}, then everything above it also equals 12 unless you override it so no need for the md tag